From 3af83b993aa850882baa31fbf4ebf832df7266c6 Mon Sep 17 00:00:00 2001
From: xzhang <xzhang@xzdt.(none)>
Date: Thu, 30 Jan 2014 10:39:20 -0800
Subject: [PATCH 261/375] CDH-15744: hive script removes AUX_CLASSPATH if HIVE_AUX_JARS_PATH is set

---
 bin/hive |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/bin/hive b/src/bin/hive
index a9feb65..4579b07 100755
--- a/src/bin/hive
+++ b/src/bin/hive
@@ -119,7 +119,7 @@ elif [ "${HIVE_AUX_JARS_PATH}" != "" ]; then
       HIVE_AUX_JARS_PATH=`cygpath -p -w "$HIVE_AUX_JARS_PATH"`
       HIVE_AUX_JARS_PATH=`echo $HIVE_AUX_JARS_PATH | sed 's/;/,/g'`
   fi
-  AUX_CLASSPATH=${HIVE_AUX_JARS_PATH}
+  AUX_CLASSPATH=${AUX_CLASSPATH}:${HIVE_AUX_JARS_PATH}
   AUX_PARAM="file://$(echo ${HIVE_AUX_JARS_PATH} | sed 's/:/,file:\/\//g')"
 fi
 
-- 
1.7.0.4

